tests/network/networkpolicy.go

```diff
@@ -228,9 +293,14 @@ var _ = SIGDescribe("[Serial][rfe_id:150][crit:high][vendor:cnv-qe@redhat.com][l
 						},
 					},
 				)
+
+				clientVMI, err = createClientVmi(virtClient, util.NamespaceTestDefault)
+				Expect(err).ToNot(HaveOccurred())
+				assertIPsNotEmptyForVMI(clientVMI)
 			})
 			AfterEach(func() {
 				waitForNetworkPolicyDeletion(policy)
+				Expect(tests.DeleteVmiAndWaitForDeletion(virtClient, clientVMI)).To(Succeed())
```

Comment:
It is not clear to me if you need to wait for the VMIs to disappear or if you just to that as a precation? In general we try to write the tests in a way that we can cope with some leftovers, since waiting for VMIs to disappear can take quite some time.

Reply:
Just a precaution. I'll go over and if it is not necessary I'll remove the wait part.

Reply:
Also, maybe if we are not waiting for the vmi to be gone, we can just rely on the global cleanup which happens after each test?

Reply:
Yes. Until all `BeforeAll` are remove from the test suite, every test file does the cleanup itself with

```
BeforeEach() {
 tests.BeforeTestCleanup()
}
```

which deletes everything but does not wait, to avoid the long wait penalties.

Reply:
Tnx:)

Reply:
maybe should be done in the AfterEach ? 
as this PR fixed
https://github.com/kubevirt/kubevirt/pull/6335

Reply:
Yes can be done in an `AfterEach` too. A the end once all `BeforeAll` are gone we just remove all calls of `tests.BeforeTestCleanup()` and move it to one global `JustAfterEach`.

Reply:
But right now you have the best guarantees if you use a `BeforeEach`.

Reply:
> maybe should be done in the AfterEach ?
as this PR fixed
#6335

IIRC currently the global cleanup with `tests.BeforeTestCleanup()` must happen before the set-up stage, because collecting debugging info on failures happens at the tear-down,  so if we clean everything at the tear-down we might miss some info.
So if I'll use an `AFfterEach` I must not rely on `tests.BeforeTestCleanup()`.

Reply:
You right, but there is this new PR that make both works
https://github.com/kubevirt/kubevirt/pull/6350

Anyhow the issue there was a private case that caused seg-fault, so whatever you choose

Reply:
> IIRC currently the global cleanup with `tests.BeforeTestCleanup()` must happen at the set-up stage, because collecting debugging info on failures happens at the tear-down, so if we clean everything at the tear-down we might miss some info.
> So if I'll use an `AFfterEach` I must not rely on `tests.BeforeTestCleanup()`.

Yes, but luckily that changed a few days ago: https://github.com/kubevirt/kubevirt/pull/6350/files. We collect now always properly.

Reply:
Nice. Thanks.